pubsub/consumer.go

```diff
@@ -0,0 +1,170 @@
+package pubsub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"github.com/offchainlabs/nitro/util/redisutil"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	"github.com/spf13/pflag"
+)
+
+type ConsumerConfig struct {
+	// Timeout of result entry in Redis.
+	ResponseEntryTimeout time.Duration `koanf:"response-entry-timeout"`
+	// Duration after which consumer is considered to be dead if heartbeat
+	// is not updated.
+	KeepAliveTimeout time.Duration `koanf:"keepalive-timeout"`
+	// Redis url for Redis streams and locks.
+	RedisURL string `koanf:"redis-url"`
+	// Redis stream name.
+	RedisStream string `koanf:"redis-stream"`
+	// Redis consumer group name.
+	RedisGroup string `koanf:"redis-group"`
+}
+
+var DefaultConsumerConfig = &ConsumerConfig{
+	ResponseEntryTimeout: time.Hour,
+	KeepAliveTimeout:     5 * time.Minute,
+	RedisStream:          "default",
```

Comment:
for both producer and consumer:
A single redis server should have multiple streams. To avoid confusion/misconfiguration - default should be empty, and producer/consumer should fail to init with empty string (like redisURL)